src/base/utils/string.cpp

```diff
@@ -65,7 +70,11 @@ QString qt_regexp_toCanonical(const QString &pattern, QRegExp::PatternSyntax pat
 
 QString Utils::String::wildcardToRegexPattern(const QString &pattern)
 {
-    return qt_regexp_toCanonical(pattern, QRegExp::Wildcard);
+    // replace [ and ] with [[] and []]
+    QString escapedPattern = pattern;
+    escapedPattern.replace(QRegExp(u"\\[|\\]"_qs), u"[\\0]"_qs);
```

Comment:
`QRegularExpression` is available and should be preferred.

Reply:
Yes, but it would require adding `QRegularExpression` to the includes above: 

```
#if (QT_VERSION >= QT_VERSION_CHECK(6, 0, 0))
#include <QRegularExpression>
#else
#include <QRegExp>
#endif
```

I thought it would be more efficient if we avoid adding that include.

Reply:
>I thought it would be more efficient if we avoid adding that include.

I don't think it is much of a problem. `QRegularExpression` is already in use elsewhere in the codebase.

I suggest you edit the headers this way:
```c++
#include <QLocale>
#include <QRegularExpression>
#include <QVector>

#if (QT_VERSION < QT_VERSION_CHECK(6, 0, 0))
#include <QRegExp>
#endif
```

Reply:
Force-pushed to include your suggestion.